File: src/main/java/rag/app/stages/PipelineStage.java
package rag.app.stages;

import rag.app.Context;
import rag.trace.TraceBus;

public interface PipelineStage {

    // Her stage’in adı: Trace logları için zorunlu.
    String getName();

    // Stage’in gerçek işi burada yapılır.
    // Context: pipeline state
    // TraceBus: log sistemi
    void run(Context context, TraceBus traceBus) throws Exception;
}

File: src/main/python/rag/app/stages/pipeline_stage.py
from typing import Protocol
from rag.app.context import Context
from rag.trace.trace_bus import TraceBus


class PipelineStage(Protocol):
    def get_name(self) -> str:
        ...

    def run(self, context: Context, trace_bus: TraceBus) -> None:
        ...






